elm-test/Main.elm

```diff
@@ -0,0 +1,90 @@
+module Main exposing (main)
+
+import Html exposing (..)
+import Html.Attributes exposing (..)
+import Html.Events exposing (onInput, onClick)
+
+type WeightUnit
+    = Pounds
+    | Kilos
+
+type alias WorkoutSet =
+    { reps : List Int
+    , weight : Int
+    , unitOfMeasurement : WeightUnit
+    }
+
+type Msg
+    = RepsUpdated String
+    | WeightUpdated String
+    | UnitUpdated WeightUnit
+
+initialWorkoutSet : WorkoutSet
+initialWorkoutSet =
+    { reps = []
+    , weight = 0
+    , unitOfMeasurement = Kilos
+    }
+
+view : WorkoutSet -> Html Msg
+view set =
+    div []
+        [ input [type_ "text", placeholder "reps", onInput RepsUpdated ] []
+        , input [type_ "text", placeholder "weight", onInput WeightUpdated] []
+        , select [ onInput (UnitUpdated << parseUnit) ]
+            [ viewUnit set.unitOfMeasurement Pounds
+            , viewUnit set.unitOfMeasurement Kilos
+            ]
```

Comment:
where `let units = [Pounds, Kilos]`? That makes sense. I think for the sake of explicitness (and because it's just 2 elements) I'll probably leave this as is, but that's good to keep in mind for the future.